### PR TITLE
Improving AutoYaST ERB documentation

### PR DESCRIPTION
**Pull request by joseivanlopez**
### PR creator: Description

Extend documentation about AutoYaST ERB templates.

### PR creator: Are there any relevant issues/feature requests?

* bsc#...
* jsc#SLE-...

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 SP4/openSUSE Leap 15.4 *(current `main`, no backport necessary)*
  - [x] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
  - [ ] SLE 15 GA
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4
  - [ ] SLE 12 SP3

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Review by imobachgs — COMMENTED** [review]
Great work. Thanks, @joseivanlopez!

I have added a few notes, but I would say that the content is fine.

File: xml/ay_erb_templates.xml [hunk 1]
```diff
@@ -465,5 +520,187 @@ xml:id="erb-templates">
...
+  <sect2 xml:id="erb-hardware-helper">
+   <title><literal>hardware</literal></title>
+   <para>
+    The information about disks through the <literal>disks</literal> helper is
+    pretty limited. The <literal>hardware</literal> helper provides additional
+    hardware information. Basically it returns all the information from
+    <literal>hwinfo</literal> command. You can use this helper as an escape hatch.
+    In the next example, <literal>hardaware</literal> helper is used to filter USB
+    devices. Check <xref linkend="erb-running-helpers" /> to learn how to inspect
+    all the information provided by <literal>hardware</literal> helper.
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply xml/ay_erb_templates.xml:533]: Why provided is wrong?

File: xml/ay_erb_templates.xml [hunk 2]
```diff
@@ -465,5 +520,187 @@ xml:id="erb-templates">
...
+</screen>
+   </example>
+  </sect2>
+ </sect1>
+
+ <sect1 xml:id="erb-running-helpers">
+  <title>Running <literal>ERB</literal> helpers</title>
+
+  <para>
+   You can use the <literal>Ruby</literal> console to run &ay; <literal>ERB</literal>
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply xml/ay_erb_templates.xml:549]: Umm, this document is already using `<literal/>` for ERB. Should I remove all occurrences?

File: xml/ay_erb_templates.xml [hunk 3]
```diff
@@ -465,5 +520,187 @@ xml:id="erb-templates">
...
+  </example>
+ </sect1>
+
+ <sect1 xml:id="erb-comparisson">
+  <title><literal>ERB</literal> Vs rules and classes</title>
+
+  <para>
+   Although both <literal>ERB</literal> and rules/classes allow to generate profiles
+   dinamically, in general <literal>ERB</literal> profiles are easier to read and
+   understand. One important difference is that rules and classes are able to merge profiles,
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply xml/ay_erb_templates.xml:647]: Yes, it is tricky.

File: xml/ay_erb_templates.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (imobachgs) [reply xml/ay_erb_templates.xml:549]: > Umm, this document is already using `<literal/>` for ERB. Should I remove all occurrences?

You are (partially) right. :-) And that's the problem, some occurrences are using literal, and some occurrences are not using it. They are mostly in the "What is ERB" section.

File: xml/ay_erb_templates.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (imobachgs) [reply xml/ay_erb_templates.xml:533]: Sorry, there was a typo in my comment. *provided by the* ("the" is missing).

**Review by cjschroder — APPROVED** [review]
This is great, thank you! There are a couple of small typos I will correct in a separate commit. Thank you very much!

**Comment by cjschroder** [comment]
small corrections: https://github.com/SUSE/doc-sle/pull/1124

**Pull request merged**